sql/memory.go

```diff
@@ -0,0 +1,198 @@
+package sql
+
+import (
+	"os"
+	"runtime"
+	"strconv"
+	"sync"
+
+	errors "gopkg.in/src-d/go-errors.v1"
+)
+
+// Disposable objects can erase all their content when they're no longer in use.
+// They should not be used again after they've been disposed.
+type Disposable interface {
+	// Dispose the contents.
+	Dispose()
+}
+
+// Freeable objects can free their memory.
+type Freeable interface {
+	// Free the memory.
+	Free()
+}
+
+// KeyValueCache is a cache of key value pairs.
+type KeyValueCache interface {
+	// Put a new value in the cache. If there is no more memory and the cache is
+	// not Freeable it will try to free some memory from other freeable caches.
+	// If there's still no more memory available, it will fail and erase all the
```

Comment:
This shouldn't be the case in all the implementations. Per example, a warmed cache can improve a lot consecutive queries execution over more or less the same data.

Reply:
Ok, will remove that from the doc